### PR TITLE
Added local source support for BareMetal platform

### PR DESCRIPTION
**Pull request by paulli2017**
Added local source support for BareMetal platform

### CODE REVIEW

File: lisa/sut_orchestrator/baremetal/source.py [hunk 1]
```diff
@@ -83,3 +83,26 @@ def extract(self, artifact_path: Path) -> str:
...
+        return "local"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return LocalSourceSchema
+
+    def download(self, timeout: int = 600) -> List[Path]:
+        downloaded_paths: List[Path] = []
+
+        for artifact in self.local_runbook.artifacts:
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/baremetal/source.py:113]: Please support extract like smb source. Move some shared code to the base class.

Comment (paulli2017) [reply lisa/sut_orchestrator/baremetal/source.py:113]: The change is done, thanks

File: lisa/sut_orchestrator/baremetal/source.py [hunk 2]
```diff
@@ -61,25 +89,31 @@ def download(self, timeout: int = 600) -> List[Path]:
...
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return LocalSourceSchema
+
+    def download(self, timeout: int = 600) -> List[Path]:
+        local_artifacts_path: List[Path] = []
+
+        for artifact in self.local_runbook.artifacts:
+            local_artifacts_path.append(Path(artifact.artifact_name))
+
+        return _process_extract(
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/baremetal/source.py]: Update the SMBSource to use the same logic.

File: lisa/sut_orchestrator/baremetal/source.py [hunk 3]
```diff
@@ -12,7 +12,35 @@
...
+def _extract(artifact_path: Path) -> str:
+    file_extension = artifact_path.suffix
+    if file_extension == ".zip":
+        with zipfile.ZipFile(str(artifact_path), "r") as zip_ref:
+            zip_ref.extractall(str(artifact_path.parent))
+    source_path = os.path.splitext(str(artifact_path))[0]
+    return source_path
+
+
+def _process_extract(
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/baremetal/source.py]: `_extract_artifacts`

File: lisa/sut_orchestrator/baremetal/source.py (hunk 2 above)
Comment (paulli2017) [reply lisa/sut_orchestrator/baremetal/source.py]: The change is already done, thanks

File: lisa/sut_orchestrator/baremetal/source.py (hunk 3 above)
Comment (paulli2017) [reply lisa/sut_orchestrator/baremetal/source.py]: The change is done, thanks

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**